nilearn/input_data/confounds_parser.py

```diff
@@ -0,0 +1,305 @@
+        of timecourses, VARS referring to root mean squared variance over
+        voxels.
+
+    wm_csf : {'basic', 'power2', 'derivatives', 'full'}
+        Type of confounds extracted from masks of white matter and
+        cerebrospinal fluids.
+
+        - "basic" the averages in each mask (2 parameters)
+        - "power2" averages and quadratic terms (4 parameters)
+        - "derivatives" averages and derivatives (4 parameters)
+        - "full" averages + derivatives + quadratic terms + power2d derivatives
+          (8 parameters)
+
+    global_signal : {'basic', 'power2', 'derivatives', 'full'}
+        Type of confounds extracted from the global signal.
+
+        - "basic" just the global signal (1 parameter)
+        - "power2" global signal and quadratic term (2 parameters)
+        - "derivatives" global signal and derivative (2 parameters)
+        - "full" global signal + derivatives + quadratic terms + power2d
+          derivatives (4 parameters)
+
+    scrub : {'full', 'basic'}
+        Type of scrub of frames with excessive motion (Power et al. 2014)
+
+        - "basic" remove time frames based on excessive framewise displacement
+          and DVARS.
+        - "full" also remove continuous segments containing fewer than 5
+          volumes.
+
+        One-hot encoding vectors are added as regressors for each scrubbed
+        frame.
+
+    compcor : {'anat_combined', 'anat_separated', 'temporal',
+    'temporal_anat_combined', 'temporal_anat_separated'}
+
+        .. warning::
+            Require fmriprep >= v:1.4.0.
+
+        Type of confounds extracted from a component based noise correction
+        method.
+
+        - "anat_combined" noise components calculated using a white matter and
+          CSF combined anatomical mask
+        - "anat_separated" noise components calculated using white matter mask
+          and CSF mask compcor separately; two set of scores are concatenated
+        - "temporal" noise components calculated using temporal compcor
+        - "temporal_anat_combined" components of "temporal" and "anat_combined"
+        - "temporal_anat_separated" components of "temporal" and
+          "anat_separated"
+
+    n_compcor : "auto" or int, default "auto"
+        The number of noise components to be extracted.
+        For acompcor_combined=False, and/or compcor="full", this is the number
+        of components per mask.
+        "auto": select all components (50% variance explained by fMRIPrep
+        defaults)
+
+    ica_aroma : {'full', 'basic'}
+
+        - "full": use fMRIprep output `~desc-smoothAROMAnonaggr_bold.nii.gz`.
+        - "basic": use noise independent components only.
+
+    demean : boolean, default True
+        If True, the confounds are standardized to a zero mean (over time).
+        This step is critical if the confounds are regressed out of time series
+        using nilearn with no or zscore standardization, but should be turned
+        off with "spc" normalization.
```

Comment:
Firstly there's a type (psc rather than spc) :sweat_smile: 
Thanks for raising this! I found a potential bind spot in how we simulate the signal for testing when looking into this.
Had a closer look at the code of `signal.clean` - the confounds will be standardised or scaled before denoising. Whether the output here is demeaned here doesn't seem to matter following the logic. 
I might need to dig into this further...

Reply:
I think this comment is deprecated since this issue has been fixed https://github.com/nilearn/nilearn/issues/2584
I think we should remove this part of the documentation. 
However I would be tempted to keep the tests which check that confounds are properly regressed with different types of normalization - just to prevent an issue with normalization from coming back in the future.

Reply:
actually the tests in load_confounds may be redundant with the ones that where added in this PR https://github.com/nilearn/nilearn/pull/2609 I haven't yet had to look in details, but we may want to avoid duplicating tests.